Application/ControlMsrE2/ControlMsrE2.h

```diff
@@ -0,0 +1,223 @@
+/** @file
+  Control CFGLock BIOS Option
+
+Copyright (c) 2020, Brumbaer. All rights reserved.<BR>
+This program and the accompanying materials
+are licensed and made available under the terms and conditions of the BSD License
+which accompanies this distribution.  The full text of the license may be found at
+http://opensource.org/licenses/bsd-license.php
+
+THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+
+**/
+
+///
+/// Unless otherwise specified all data types are naturally aligned. Structures are
+/// aligned on boundaries equal to the largest internal datum of the structure and
+/// internal data are implicitly padded to achieve natural alignment.
+///
+
+#include <Uefi.h>
+#include <PiDxe.h>
+#include <Library/PcdLib.h>
+#include <Library/UefiLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/OcMiscLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <Library/UefiApplicationEntryPoint.h>
+
+#include <Library/MemoryAllocationLib.h>
+#include <Uefi/UefiInternalFormRepresentation.h>
+#include <Library/UefiHiiServicesLib.h>
+#include <Library/HiiLib.h>
+
+#include <Protocol/HiiFont.h>
+#include <Protocol/HiiImage.h>
+#include <Protocol/HiiImageEx.h>
+#include <Protocol/HiiImageDecoder.h>
+#include <Protocol/HiiString.h>
+#include <Protocol/HiiDatabase.h>
+#include <Protocol/HiiConfigRouting.h>
+#include <Protocol/HiiConfigAccess.h>
+#include <Protocol/HiiConfigKeyword.h>
+#include <Guid/HiiPlatformSetupFormset.h>
+
+#define DONT_STOP_AT 0xFFFF
+
+/**
+  Parameters to search in a Form for
+  IfrHeaders which contain a Description
+  and to return the VarStore
+
+  @param SearchText        Text that must be part of the options description
+  @param EfiHandle         Handle of the list the Form is part of
+  @param ListHeader        Ptr to the contents of the handle
+  @param PkgHeader         Ptr to the current package in the list
+  @param FirstIfrHeader    Ptr to first IfrHeader in Package
+  @param IfrVarStore       Ptr to IfrHeader of corresponding varstore in same list
+  @param IfrOneOf          Ptr to IfrHeader of BIOS Option
+  @param StopAt            Used to find a certain BIOS Option
+  @param Count             Running number of suitable BIOS Options
```

Comment:
Not sure how Count exactly works, but it generally is UINTN or UINT32.

Reply:
There is a `#define CONTEXTS_MAX 8` limitation in `ControlMsrE2.c` for the first 8 results obtained from HiiDatabase. So this parameter won't be that large?